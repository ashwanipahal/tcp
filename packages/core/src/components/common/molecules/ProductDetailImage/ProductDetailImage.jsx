/* eslint-disable */
// Temp fix for DAM image in PLP
import React from 'react';
import PropTypes from 'prop-types';
import ExecutionEnvironment from 'exenv';
import ReactImageMagnify from 'react-image-magnify';
import RenderPerf from '@tcp/web/src/components/common/molecules/RenderPerf';
import { HERO_VISIBLE } from '@tcp/core/src/constants/rum.constants';
import { DamImage, Anchor } from '../../atoms';
import withStyles from '../../hoc/withStyles';
import styles from './ProductDetailImage.style';
import { getLocator } from '../../../../utils';

const getNonZoomImage = (isMobile, imageUrl, imageName, onOpenSimpleFullSize) => {
  const imgData = {
    alt: imageName,
    url: imageUrl,
  };
  return !isMobile ? (
    <DamImage
      className="full-size-desktop-image"
      data-locator={getLocator('pdp_main_image')}
      imgData={imgData}
      itemProp="contentUrl"
      isProductImage
    />
  ) : (
    <Anchor aria-label="view full size image" onClick={onOpenSimpleFullSize}>
      <DamImage
        className="full-size-desktop-image"
        data-locator={getLocator('pdp_main_image')}
        imgData={imgData}
        itemProp="contentUrl"
        isProductImage
      />
    </Anchor>
  );
};

const ProductDetailImage = props => {
  const {
    imageName,
    imageUrl,
    zoomImageUrl,
    className,
    isZoomEnabled,
    onOpenSimpleFullSize,
    isMobile,
  } = props;

  let productSectionWidth;

  if (ExecutionEnvironment.canUseDOM) {
    productSectionWidth =
      document.getElementById('productDetailsSection') &&
      document.getElementById('productDetailsSection').offsetWidth;
  }
  return (
    <div itemScope itemType="http://schema.org/ImageObject" className={className} title={imageName}>
      {/* {isZoomEnabled && !isMobile ? (
        <ReactImageMagnify
          data-locator={getLocator('pdp_main_image')}
          {...{
            enlargedImagePortalId: 'portal',
            smallImage: {
              src: imageUrl,
              isFluidWidth: true,
              alt: imageName,
            },
            enlargedImageContainerDimensions: {
              width: productSectionWidth || '100%',
              height: '100%',
            },
            largeImage: {
              src: zoomImageUrl,
              width: 900,
              height: 900,
              alt: imageName,
            },
            enlargedImageContainerClassName: 'enlarged-image-wrapper',
            isActivatedOnTouch: false,
          }}
        />
      ) : (
        getNonZoomImage(isMobile, imageUrl, imageName, onOpenSimpleFullSize)
      )} */}
      {getNonZoomImage(isMobile, imageUrl, imageName, onOpenSimpleFullSize)}
      <RenderPerf.Measure name={HERO_VISIBLE} />
    </div>
  );
};

ProductDetailImage.propTypes = {
  imageName: PropTypes.string,

  /** the image url */
  imageUrl: PropTypes.string.isRequired,
  isMobile: PropTypes.bool.isRequired,

  /** zoom image url */
  zoomImageUrl: PropTypes.string.isRequired,

  className: PropTypes.string,
  isZoomEnabled: PropTypes.bool,
  onOpenSimpleFullSize: PropTypes.func,
};

ProductDetailImage.defaultProps = {
  className: '',
  imageName: '',
  isZoomEnabled: true,
  onOpenSimpleFullSize: () => {},
};

export default withStyles(ProductDetailImage, styles);
export { ProductDetailImage as ProductDetailImageVanilla };
