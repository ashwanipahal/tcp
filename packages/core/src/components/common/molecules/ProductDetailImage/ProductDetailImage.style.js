import { css } from 'styled-components';

const styles = css`
  img {
    max-height: 427px;
  }

  .full-size-desktop-image {
    max-height: none;
    width: 100%;
  }

  @media ${props => props.theme.mediaQuery.medium} {
    img {
      max-height: 436px;
    }
  }
  @media ${props => props.theme.mediaQuery.large} {
    img {
      max-height: 703px;
    }
  }
`;

export default styles;
