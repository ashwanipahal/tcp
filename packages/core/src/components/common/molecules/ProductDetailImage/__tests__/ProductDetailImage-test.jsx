import React from 'react';
import { shallow } from 'enzyme';

import { ProductDetailImageVanilla } from '../ProductDetailImage';

describe('ProductDetailImage component', () => {
  it('should renders correctly', () => {
    const props = {
      imageName: '',
      imageUrl: '',
      zoomImageUrl: '',
      className: '',
    };
    const component = shallow(<ProductDetailImageVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
});
