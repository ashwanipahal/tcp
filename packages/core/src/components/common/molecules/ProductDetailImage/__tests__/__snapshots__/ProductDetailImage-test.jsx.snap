// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailImage component should renders correctly 1`] = `
<div
  className=""
  itemScope={true}
  itemType="http://schema.org/ImageObject"
  title=""
>
  <ReactImageMagnify
    data-locator="pdp_main_image"
    enlargedImageContainerClassName="enlarged-image-wrapper"
    enlargedImageContainerDimensions={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    enlargedImagePortalId="portal"
    fadeDurationInMs={300}
    hintComponent={[Function]}
    hintTextMouse="Hover to Zoom"
    hintTextTouch="Long-Touch to Zoom"
    hoverDelayInMs={250}
    hoverOffDelayInMs={150}
    isActivatedOnTouch={false}
    isEnlargedImagePortalEnabledForTouch={false}
    isHintEnabled={false}
    largeImage={
      Object {
        "alt": "",
        "height": 900,
        "src": "",
        "width": 900,
      }
    }
    shouldHideHintAfterFirstActivation={true}
    shouldUsePositiveSpaceLens={false}
    smallImage={
      Object {
        "alt": "",
        "isFluidWidth": true,
        "onLoad": [Function],
        "src": "",
      }
    }
  />
</div>
`;
